isis/src/base/objs/ImageHistogram/ImageHistogram.cpp

```diff
@@ -0,0 +1,230 @@
+ *   http://www.usgs.gov/privacy.html.
+ */
+#include "ImageHistogram.h"
+
+#include "ControlNet.h"
+#include "ControlMeasure.h"
+
+#include <iostream>
+#include <math.h>
+#include <stdio.h>
+#include <string>
+
+using namespace std;
+
+namespace Isis {
+
+  /**
+   * Constructs a histogram object. Only data between the minimum and maximum
+   * will be binned, and the bin range will be from the minimum to the maximum.
+   *
+   * @param minimum Minimum value for binning the data into the histogram.
+   * @param maximum Maximum value for binning the data into the histogram.
+   * @param nbins The number of bins to use
+   */
+  ImageHistogram::ImageHistogram(double minimum, double maximum, int nbins) :
+  Histogram(minimum, maximum, nbins) {
+  }
+
+
+  /**
+   * Constructs a histogram object using a cube. This constructor computes
+   *   the minimum, maximum for the binning range and number of bins
+   *   automatically. All statistics will still be collected, though data at
+   *   either end of the histogram will be put into one bin in order to attempt to
+   *   achieve better histogram statistics.
+   *
+   * TODO: progress needs to be a bool.
+   *
+   * @param cube  The cube to used to determine min/max and bins
+   * @param statsBand  The band number the histogram will be collected from
+   * @param progress  The Progress object to be used to output the percent
+   *                  processed information
+   * @param startSample  The sample to start reading cube data from
+   * @param startLine  The line to start reading cube data from
+   * @param endSample  The sample to stop reading cube data at (Null for nsamps)
+   * @param endLine  The line to stop reading cube data at (Null for nlines)
+   * @param bins The number of histogram bins to create (0 for automatic)
+   * @param addCubeData True to fill the histogram with data in addition to
+   *                    initializing the binning ranges.
+   */
+  ImageHistogram::ImageHistogram(Cube &cube, int statsBand, Progress *progress,
+      double startSample, double startLine,
+      double endSample, double endLine,
+      int bins, bool addCubeData) :
+  Histogram(cube, statsBand, progress,
+                startSample, startLine,
+                endSample, endLine,
+                bins, addCubeData) {
+  }
+
+
+  /**
+   * Constructs a histogram from a control netowrk
+   *
+   * @param net: Reference to a ControlNetwork used to access all the measures.
+   * @param statFunc: Pointer to a ControlMeasure acessor, the returns of this
+   *    function call will be used to build up the network.
+   * @param bins:  The number of bins to divide the histogram into.
+   * @throws The number of Histogram Bins must be greater than 0.
+   */
+  ImageHistogram::ImageHistogram(ControlNet &net, double(ControlMeasure::*statFunc)() const, int bins) :
+  Histogram(net, statFunc, bins) {
+
+    //add all the data to the now setup histogram
+    addMeasureDataFromNet(net, statFunc);
```

Comment:
Is this why the control network constructor doesn't add the data in the Histogram class anymore?